### PR TITLE
Spawn with pipes

### PR DESCRIPTION
**Pull request by psychon**
This adds `awful.util.spawn_with_lines` (TODO: Suggestions for a better name?) that spawns a program and captures its standard output line by line, calling a lua function on each line. Additionally, another function is called when this pipe is closed.

Ref: https://github.com/awesomeWM/awesome/pull/378

### CODE REVIEW

**Comment by psychon** [comment]
I did some experiments and now added a `tostring`. Apparently older lgi versions gave us an instance of GLib.Bytes or something like that instead of a string. This should fix all CI failures.

**Comment by blueyed** [comment]
Awesome! Thanks!

File: lib/awful/util.lua [hunk 1]
```diff
@@ -110,6 +112,61 @@ function util.spawn_with_shell(cmd, sn)
...
+--- Spawn a program and asynchronously capture its output line by line.
+-- @tparam string|table cmd The command.
+-- @tparam function line_callback Function to call with each line of output.
+-- @tparam[opt] function done_callback Function to call when no more output is
+-- produced.
+-- @treturn[1] integer The forked PID.
+-- @treturn[2] string Error message.
+function util.spawn_with_lines(cmd, line_callback, done_callback)
+    -- Spawn without startup notification and with returning a pipe for stdout
+    local pid, _, _, stdout, _ = capi.awesome.spawn(cmd, false, false, true, false)
```
Comment (blueyed) [inline lib/awful/util.lua]: I wonder if `stderr` should also be captured and passed to the callback function?
Could `start_read` read both stdout and stderr, and call the callback accordingly from what was read from?

File: lib/awful/util.lua [hunk 2]
```diff
@@ -110,6 +112,61 @@ function util.spawn_with_shell(cmd, sn)
     end
 end
 
+--- Spawn a program and asynchronously capture its output line by line.
+-- @tparam string|table cmd The command.
+-- @tparam function line_callback Function to call with each line of output.
```
Comment (blueyed) [inline lib/awful/util.lua]: Please document the signature of the callback function, e.g. that it gets `line_from_stdout`.

**Comment by blueyed** [comment]
Waiting for an update.. please move to "next" then.

File: lib/awful/util.lua [hunk 3]
```diff
@@ -110,6 +112,94 @@ function util.spawn_with_shell(cmd, sn)
     end
 end
 
+--- Spawn a program and asynchronously capture its output line by line.
```
Comment (blueyed) [inline lib/awful/util.lua]: Missing an "and".  I will amend that.

**Pull request merged**